week_4/quiz-trivia-app/src/app/header/header.component.ts

```diff
@@ -0,0 +1,42 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { Quiz } from '../data/quiz.model';
+import { StorageService } from '../storage.service';
+
+@Component({
+  selector: 'app-header',
+  standalone: true,
+  imports: [],
+  templateUrl: './header.component.html',
+  styleUrl: './header.component.css',
+})
+export class HeaderComponent implements OnInit {
+  @Input() subject: Quiz | undefined = undefined;
+  isDarkMode = false;
+
+  constructor(private storageService: StorageService) {}
+
+  ngOnInit(): void {
+    const savedTheme = localStorage.getItem('theme');
+    if (savedTheme) {
+      this.isDarkMode = savedTheme === 'dark-mode';
+      document.body.classList.toggle(savedTheme);
+    }
+
+    const storedSubject = this.storageService.getData<Quiz>('selectedSubject');
```

Comment:
You dont need this variable, you can directly assign this.subject when u get from storage. The if statement inst necessary, because even if getData doesnt return anything, You dont really do anything so just assign directly

Reply:
done